lw/net/internal/tls_context.cpp

```diff
@@ -0,0 +1,127 @@
+#include "lw/net/internal/tls_context.h"
+
+#include <experimental/source_location>
+#include <memory>
+#include <string_view>
+
+#include "lw/err/canonical.h"
+#include "lw/err/system.h"
+#include "openssl/bio.h"
+#include "openssl/crypto.h"
+#include "openssl/err.h"
+#include "openssl/ssl.h"
+
+namespace lw::net::internal {
+namespace {
+
+using std::experimental::source_location;
+
+void check_openssl_error(source_location loc = source_location::current()) {
+  auto err_code = ERR_get_error();
+  if (!err_code) return;
+
+  // OpenSSL recommends a minimum of 256 bytes for this buffer.
+  char err[1024] = {0};
+  ERR_error_string_n(err_code, err, sizeof(err));
+
+  // TODO(alaina): Split OpenSSL errors by class of error instead of all being
+  // Internal.
+  throw Internal(loc) << "OpenSSL error: " << static_cast<char*>(err);
+}
+
+[[noreturn]] void check_all_errors(
```

Comment:
Unify `check_openssl_error` and `check_all_errors` between the `TLSContextImpl` and `TLSClientImpl` files in a `check_errors` unit.